fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -311,33 +313,46 @@ Future<Void> replicaComparison(Req req,
 			}
 		} else if (!srcLB.present() || !srcLB.get().error.present()) {
 			// Verify that the other SS servers in the team have the same data.
-			state std::vector<Future<Optional<ErrorOr<Resp>>>> restOfTeamFutures;
-			restOfTeamFutures.reserve(ssTeam->size() - 1);
+			std::vector<uint64_t> candidates;
+			// candidates includes all healthy SS endpoints in the team except the one we already
+			// have a response from
 			for (int i = 0; i < ssTeam->size(); i++) {
 				RequestStream<Req, P> const* si = &ssTeam->get(i, channel);
-				if (si->getEndpoint().token.first() !=
-				    srcEndpointId) { // don't re-request to SS we already have a response from
-					if (!IFailureMonitor::failureMonitor().getState(si->getEndpoint()).failed) {
-						resetReply(req);
-						restOfTeamFutures.push_back((
-						    requiredReplicas == BEST_EFFORT
-						        ? timeout(si->tryGetReply(req), FLOW_KNOBS->LOAD_BALANCE_FETCH_REPLICA_TIMEOUT)
-						        : timeout(errorOr(si->getReply(req)), FLOW_KNOBS->LOAD_BALANCE_FETCH_REPLICA_TIMEOUT)));
-					} else if (requiredReplicas == ALL_REPLICAS) {
-						TraceEvent(SevWarnAlways, "UnreachableStorageServer")
-						    .detail("SSID", ssTeam->getInterface(i).id());
-						throw unreachable_storage_replica();
-					}
+				if (si->getEndpoint().token.first() == srcEndpointId) {
```

Comment:
Is this mainly code refactoring? Or is there a real issue that you are trying to solve here?

Reply:
No this is not a refactoring, the new code changes the logic. In the old code, we are doing quorum reads, which causes sending requests to all replica. In the new code, we only send requests to numReplicaToRead SSes. The numReplicaToRead SSes are randomly selected. To achieve this, we have the new code here.